dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/deduplication/PotentialDuplicatesTests.java

```diff
@@ -29,99 +29,90 @@
-        response
+        potentialDuplicatesActions.update( duplicateId + "?status=" + "INVALID", new JsonObjectBuilder().build() )
             .validate()
-            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
-            .body( "identifiableObjects.status", everyItem( equalTo( status ) ) );
-    }
+            .statusCode( 200 );
 
-    @Test
-    public void shouldReturnAllStatuses()
-    {
-        Arrays.asList( "OPEN", "MERGED", "INVALID" ).forEach( status -> {
-            String teiA = createTei();
-            String teiB = createTei();
+        // MERGED
+        String potentialDuplicate = potentialDuplicatesActions.createAndValidatePotentialDuplicate( createTei(),
+            createTei() );
 
-            potentialDuplicatesActions.createPotentialDuplicate( teiA, teiB, status ).validate().statusCode( 200 );
-        } );
+        potentialDuplicatesActions.autoMergePotentialDuplicate( potentialDuplicate ).validate()
+            .statusCode( 200 );
 
         potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=ALL" ) )
             .validate()
-            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 3 ) ) )
             .body( "identifiableObjects.status",
                 allOf( hasItem( "OPEN" ), hasItem( "INVALID" ), hasItem( "MERGED" ) ) );
+
+        potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + "OPEN" ) )
+            .validate()
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects.status", everyItem( equalTo( "OPEN" ) ) );
+
+        potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + "INVALID" ) )
+            .validate()
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects.status", everyItem( equalTo( "INVALID" ) ) );
+
+        potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + "MERGED" ) )
+            .validate()
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects.status", everyItem( equalTo( "MERGED" ) ) );
+
     }
 
     @Test
     public void shouldRequireBothTeis()
     {
-        potentialDuplicatesActions.createPotentialDuplicate( null, createTei(), "OPEN" )
+        potentialDuplicatesActions.createPotentialDuplicate( null, createTei() )
             .validate()
             .statusCode( equalTo( 400 ) )
             .body( "status", equalTo( "ERROR" ) )
             .body( "message", containsStringIgnoringCase( "missing required input property" ) );
     }
 
-    @CsvSource( {
-        "MERGED,INVALID,false",
-        "OPEN,INVALID,true",
-        "OPEN,MERGED,false",
-        "INVALID,OPEN,true",
-        "MERGED,OPEN,false"
-    } )
+    @CsvSource( { "MERGED" } )
```

Comment:
If your parametrized test only has one value, make it a simple test like you did in the above test.

Reply:
We lost some coverage here again and yes, if there's only value there's no need for parametrization